src/execution/vm/vm.cpp

```diff
@@ -2224,6 +2224,41 @@ void VM::Interpret(const uint8_t *ip, Frame *frame) {  // NOLINT
 #undef BINARY_REAL_MATH_OP
 #undef UNARY_REAL_MATH_OP
 
+#define ATOMIC_BINARY_OP(AOP, T)                       \
+  OP(Atomic##AOP) : {                                  \
+    auto *dest = frame->LocalAt<T *>(READ_LOCAL_ID()); \
+    auto val = frame->LocalAt<T>(READ_LOCAL_ID());     \
+    auto *ret = frame->LocalAt<T *>(READ_LOCAL_ID());  \
+    OpAtomic##AOP(dest, val, ret);                     \
```

Comment:
The semantics we've been using for all operations is that the first argument is the return argument.